tests/SffsFileApi.device.test.nut

```diff
@@ -0,0 +1,272 @@
+    local payload = file.read(buffer.len());
+    this.assertEqual(buffer.len(), file.tell());
+    this.assertEqual(file.eof(), true);
+    file.seek(0);
+    this.assertEqual(file.tell(), 0);
+    file.write(buffer);
+    file.seek(file.len());
+    this.assertTrue(file.eof());
+    this.assertEqual(file.tell(), buffer.len() * 2);
+    file.seek(file.len() - 1);
+    this.assertTrue(!file.eof());
+
+    file.close();
+    // there no way to read data from closed file
+    this.assertEqual(file.seek(0).tell(), 0);
+    // try to read data from closed file
+    payload = file.read(buffer.len());
+    this.assertEqual(file.tell(), buffer.len());
+    this.assertEqual(file.seek(file.len()).tell(), file.len());
+    this.assertTrue(file.eof());
+  }
+
+
+  // seek, tell eof after file close
+  function test05_fileClose() {
+    local file = sffs.open("test.txt", "r");
+    file.close();
+
+    this.assertEqual(file.tell(), 0);
+    // there no way to read data from closed file
+    this.assertEqual(file.seek(0).tell(), 0);
+    // try to read data from closed file
+    local payload = file.read(buffer.len());
+    this.assertEqual(file.tell(), buffer.len());
+    this.assertEqual(file.seek(file.len()).tell(), file.len());
+    this.assertTrue(file.eof());
+  }
+
+  //
+  // concurrent access to the last available free or erazed page
+  //
+  // what should happen when there is only one free
+  // page left but we are writing 2 file which need
+  // a new free page?
+  function test06_concurentAccessToTheLastFreePage() {
+    sffs.eraseFiles();
+    for (local i = 0; i < this.pages - 3; ++i)
+      createFile("file" + i + ".txt", "some payload");
+    local stats = sffs._fat.getStats();
+    // expecting that only 3 pages are available
+    this.assertEqual(3, stats.free + stats.erased);
+    //
+    // Length of the filenames are different to prevent
+    // writing of the same payload at the same
+    // address
+    local filename = "test2.txt";
+    local file1 = sffs.open("test1_diff.txt", "w");
+    local file2 = sffs.open(filename, "w");
+    // each file has heard therefore it is enough
+    // to write page size payload to each file
+    // to create an concurent access use-case
+    for (local i = 0; i < SPIFLASHFILESYSTEM_SECTOR_SIZE / buffer.len(); ++i) {
+      file1.write(buffer);
+      // an attemption to request next sector
+      try {
+        file2.write(buffer);
+      } catch (e) {
+        local expectedSize = SPIFLASHFILESYSTEM_SECTOR_SIZE -
+          maxHeaderSize - filename.len() + 1;
+        // Trying to write more data then available space
+        this.assertTrue((i + 1) * buffer.len() > expectedSize);
+        // Check that number of written data less then available space
+        this.assertTrue(file2.len() < expectedSize);
+      }
+    }
+    file1.close();
+    file2.close();
+  }
+
+  // Check filesize after the previous test which
+  // should be finished with exception
+  function test07_checkFileSize() {
```

Comment:
Usually it's not the best idea to make unit tests dependent on the order of execution. It would probably make sense to make this test a part of the previous one.